src/util/sharing_map.h

```diff
@@ -995,70 +926,16 @@ SHARING_MAPT2(const, find_type)::find(const key_type &k, const tvt &key_exists)
 /// - Best case: O(H)
 ///
 /// \param k: The key of the element to search
-/// \return Pair of const reference to found value (or dummy value if not
-///   found), and boolean indicating if element was found.
-SHARING_MAPT2(const, const_find_type)::find(const key_type &k) const
+/// \return optionalt containing a const reference to the value if found
+SHARING_MAPT2(optionalt<std::reference_wrapper<const, mapped_type>>)::find(
+  const key_type &k) const
 {
   const leaft *lp = get_leaf_node(k);
 
   if(lp == nullptr)
-    return const_find_type(dummy, false);
-
-  return const_find_type(lp->get_value(), true);
-}
-
-/// Get element at key
-///
-/// Complexity:
-/// - Worst case: O(H * S + M)
-/// - Best case: O(H)
-///
-/// \param k: The key of the element
-/// \param key_exists: Hint to indicate whether the element is known to exist
-///   (possible values `unknown` or `true`)
-/// \throw `std::out_of_range` if key not found
-/// \return The mapped value
-SHARING_MAPT2(, mapped_type &)::at(
-  const key_type &k,
-  const tvt &key_exists)
-{
-  find_type r=find(k, key_exists);
-
-  if(!r.second)
-    throw std::out_of_range(not_found_msg);
+    return nullopt;
```

Comment:
I think we had a discussion elsewhere: `return {};` appears to be more common than the use of `nullopt`.